braindecode/models/tcn.py

```diff
@@ -0,0 +1,142 @@
+                    stride=1,
+                    dilation=dilation_size,
+                    padding=(kernel_size - 1) * dilation_size,
+                    drop_prob=drop_prob
+                ))
+        self.temporal_blocks = t_blocks
+        self.fc = nn.Linear(in_features=n_filters, out_features=n_outputs)
+        if add_log_softmax:
+            self.log_softmax = nn.LogSoftmax(dim=1)
+        self.squeeze = Expression(squeeze_final_output)
+
+        self.min_len = 1
+        for i in range(n_blocks):
+            dilation = 2 ** i
+            self.min_len += 2 * (kernel_size - 1) * dilation
+
+        # start in eval mode
+        self.eval()
+
+    def forward(self, x):
+        # x is in format: B x C x T x 1
+        (batch_size, _, time_size, _) = x.size()
+        assert time_size >= self.min_len
+        # remove empty trailing dimension
+        x = x.squeeze(3)
+        x = self.temporal_blocks(x)
+        # Convert to: B x T x C
+        x = x.transpose(1, 2).contiguous()
+
+        fc_out = self.fc(x.view(batch_size * time_size, x.size(2)))
+        if hasattr(self, "log_softmax"):
+            fc_out = self.log_softmax(fc_out)
+        fc_out = fc_out.view(batch_size, time_size, fc_out.size(1))
+
+        out_size = 1 + max(0, time_size - self.min_len)
+        out = fc_out[:, -out_size:, :].transpose(1, 2)
+        # re-add 4th dimension for compatibility with braindecode
+        return self.squeeze(out[:, :, :, None])
+
+
+class TemporalBlock(nn.Module):
+    def __init__(self, n_inputs, n_outputs, kernel_size, stride, dilation,
+                 padding, drop_prob):
+        super().__init__()
+        self.conv1 = weight_norm(nn.Conv1d(
+            n_inputs, n_outputs, kernel_size,
+            stride=stride, padding=padding, dilation=dilation))
+        self.chomp1 = Chomp1d(padding)
+        self.relu1 = nn.ReLU()
+        self.dropout1 = nn.Dropout2d(drop_prob)
+
+        self.conv2 = weight_norm(nn.Conv1d(
+            n_outputs, n_outputs, kernel_size,
+            stride=stride, padding=padding, dilation=dilation))
+        self.chomp2 = Chomp1d(padding)
+        self.relu2 = nn.ReLU()
+        self.dropout2 = nn.Dropout2d(drop_prob)
+
+        self.downsample = (nn.Conv1d(n_inputs, n_outputs, 1)
+                           if n_inputs != n_outputs else None)
+        self.relu = nn.ReLU()
+
+        init.normal_(self.conv1.weight, 0, 0.01)
+        init.normal_(self.conv2.weight, 0, 0.01)
+        if self.downsample is not None:
+            init.normal_(self.downsample.weight, 0, 0.01)
+
+    def forward(self, x):
+        out = self.conv1(x)
+        out = self.chomp1(out)
+        out = self.relu1(out)
+        out = self.dropout1(out)
+        out = self.conv2(out)
+        out = self.chomp2(out)
+        out = self.relu2(out)
+        out = self.dropout2(out)
+        res = x if self.downsample is None else self.downsample(x)
+        return self.relu(out + res)
+
+
+class Chomp1d(nn.Module):
```

Comment:
Is there a better name for this module? "to chomp" means "mampfen" or "kauen" in german which is really a precise description.

Reply:
well since it is from original code let's keep it